v1.1/certificate-rotation.md

```diff
@@ -0,0 +1,178 @@
+
+While establishing an inter-node or client-node communication, the nodes and clients use the CA certificate to verify each other's identity. 
+
+For more information, see [Create security certificates](create-security-certificates.html).
+
+## When to Rotate Certificates
+
+You may need to rotate the node, client, or CA certificates in the following scenarios:
+
+- The node, client, or CA certificates are expiring soon. 
+- Your organization's compliance policy may require periodical certificate rotation.
+- The key (for the node, client, or CA) is compromised.
+- You need to modify the contents of a certificate (example: add another DNS name/IP address the node can be reached through, such as a load balancer). In this case, you need to rotate only the node certificates.
+
+## Rotating Client Certificates
+
+Rotating a client certificate involves the following steps:
+
+#### Step 1: Generating a new client key and certificate
+
+~~~ shell
+$ cockroach cert create-client <username>
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key
+~~~
+
+#### Step 2: Uploading the new client certificate and keys to the client
+
+Upload the new certificate to the client using your preferred method.
+
+#### Step 3: Have the client use the new certificate
+
+This step is application-specific and may require restarting the client.
+
+
+## Rotating Node Certificates 
+
+Rotating a node certificate requires generating a new node certificate and reloading the certificate on the node. Reloading the certificate involves uploading the node certificate to the node and rescanning the certs directory and using the new node certificate in the directory:
+
+#### Step 1: Generate new node certificate and key
+
+Suppose your existing CA and node certificates are created in the `certs` directory and your CA key is stored in a directory called `my-safe-directory`. Because the node certificate and key already exist, use the `--overwrite` flag to overwrite both certificate and key. Also specify all addresses at which node can be reached:
+
+~~~ shell
+$ cockroach cert create-node \
+[node-hostname] \
+[node-other-hostname] \
+[node-yet-another-hostname] \
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key \
+--overwrite
+~~~ 
+	
+#### Step 2: Upload the certificate and key to the node
+
+~~~ shell
+# Upload the node certificate and key:
+$ scp certs/node.crt \
+certs/node.key \
+<username>@<node1 address>:~/certs
+~~~
+
+#### Step 3: Reload the node certificate
+
+To make a running node rescan the certificates directory and use the new certificates without restarting the node, issue a `SIGHUP` signal to the cockroach process:
+	
+~~~ shell
+pkill -SIGHUP -x cockroach
+~~~
+
+The `SIGHUP` signal must be sent by the same user running the process (for example: run with `sudo` if cockroach is running under user `root`).	
```

Comment:
@mberhault, is there a way to validate that the new certificate is in use? That might be a final step here.

Reply:
Two ways:
1. look at the certificate report on the node (https://omega.crdb.io:8080/#/reports/certificates/1 for node 1)
1. look at monitoring (we should have a section on that)

Reply:
I don't think we expose the certificate report to the users. 
Also, is this the monitoring section you are referring to: https://www.cockroachlabs.com/docs/stable/create-security-certificates.html#list-certificates-and-keys

Reply:
It's linked to from the `/debug/` page, assuming access is allowed (by a cluster setting).
By monitoring, I mean the actual metrics: `security_certificate_expiration_node` and `security_certificate_expiration_ca`